### PR TITLE
Fix more <a> color

### LINKED ISSUE #91175: Invalid color used for titles' anchor
As you can see from this screenshot:

Screenshot from 2021-11-24 11-15-22

It is a regression from #91102.

cc @jsha

### PR DESCRIPTION
**Pull request by GuillaumeGomez**
Fixes #91175.

Another bug I saw is:

Screenshot from 2021-11-24 11-41-27

I fixed it as well.

r? @jsha

### CODE REVIEW

File: src/test/rustdoc-gui/anchors.goml [hunk 1]
```diff
@@ -1,17 +1,29 @@
+// This test is to ensure that the anchors (`§`) have the expected color.
 goto: file://|DOC_PATH|/test_docs/struct.HeavilyDocumentedStruct.html
 
+// This is needed to ensure that the text color is computed.
+show-text: true
```
Comment (GuillaumeGomez) [inline src/test/rustdoc-gui/anchors.goml:5]: This was originally missing, meaning that the colors weren't computed. I'll add a failure in `browser-ui-test` directly.

File: src/test/rustdoc-gui/anchors.goml [hunk 2]
```diff
@@ -1,17 +1,29 @@
...
 reload:
 
-assert-css: ("#toggle-all-docs", {"color": "rgba(0, 0, 0, 0)"})
-assert-css: (".fqn .in-band a:nth-of-type(1)", {"color": "rgba(0, 0, 0, 0)"})
-assert-css: (".fqn .in-band a:nth-of-type(2)", {"color": "rgba(0, 0, 0, 0)"})
-assert-css: (".srclink", {"color": "rgba(0, 0, 0, 0)"})
-assert-css: (".srclink", {"color": "rgba(0, 0, 0, 0)"})
+assert-css: ("#toggle-all-docs", {"color": "rgb(0, 0, 0)"})
+assert-css: (".fqn .in-band a:nth-of-type(1)", {"color": "rgb(0, 0, 0)"})
+assert-css: (".fqn .in-band a:nth-of-type(2)", {"color": "rgb(173, 68, 142)"})
```
Comment (GuillaumeGomez) [inline src/test/rustdoc-gui/anchors.goml:14]: Therefore this color was updated as well (it's for a struct so not black contrary to the other members of the import).

**Comment by jsha** [comment]
Thanks for the fix!

@bors r+

**Comment by jsha** [comment]
Per #91196 it looks like we also need a rule for `a.anchor` to be black / unstyled generally. Do you want to mix that into this PR or should I make a followup?

**Comment by camelid** [comment]
This PR is already in a rollup, so a separate PR is probably best.

**Comment by GuillaumeGomez** [comment]
Also please open an issue so we can keep track of things more easily.

**Comment by camelid** [comment]
> Also please open an issue so we can keep track of things more easily.

There's already #91196...?

**Comment by GuillaumeGomez** [comment]
Indeed... Sorry, tired. Please ignore my comment.

**Comment by jsha** [comment]
It turns out `a.anchor` is already part of this PR and I forgot! Thanks for spotting that already. :-)

**Comment by camelid** [comment]
> It turns out `a.anchor` is already part of this PR and I forgot! Thanks for spotting that already. :-)

So should #91196 be closed now?

**Pull request merged**